### PR TITLE
Enable statitcheck at build time.

### PR DESCRIPTION
**Pull request by SuperQ**
* Add `staticcheck` to the Makefile.
* Fix `staticcheck` errors.
* Ignore deprecated function in kubernetes API.

Closes: https://github.com/prometheus/prometheus/issues/3124

### CODE REVIEW

File: storage/local/mapper_test.go [hunk 1]
```diff
@@ -62,7 +62,7 @@ func TestFPMapper(t *testing.T) {
 	p, closer := newTestPersistence(t, 1)
 	defer closer.Close()
 
-	mapper, err := newFPMapper(sm, p)
+	mapper, _ := newFPMapper(sm, p)
```
Comment (grobie) [inline storage/local/mapper_test.go]: We should not ignore the error but t.Fatal on it.

File: storage/local/persistence_test.go [hunk 2]
```diff
@@ -135,7 +135,7 @@ func testPersistLoadDropChunks(t *testing.T, encoding chunk.Encoding) {
 			}
 		}
 		// Load all chunk descs.
-		actualChunkDescs, err := p.loadChunkDescs(fp, 0)
+		actualChunkDescs, _ := p.loadChunkDescs(fp, 0)
```
Comment (grobie) [inline storage/local/persistence_test.go]: Let's not ignore the error.

File: storage/remote/client_test.go [hunk 3]
```diff
@@ -64,7 +64,7 @@ func TestStoreHTTPErrorHandling(t *testing.T) {
 			panic(err)
 		}
 
-		c, err := NewClient(0, &clientConfig{
+		c, _ := NewClient(0, &clientConfig{
```
Comment (grobie) [inline storage/remote/client_test.go]: Let's not ignore the error.

File: storage/remote/queue_manager.go [hunk 4]
```diff
@@ -249,7 +249,7 @@ func (t *QueueManager) Stop() {
 func (t *QueueManager) updateShardsLoop() {
 	defer t.wg.Done()
 
-	ticker := time.Tick(shardUpdateDuration)
+	ticker := time.NewTicker(shardUpdateDuration)
```
Comment (grobie) [inline storage/remote/queue_manager.go]: Missing a defer ticker.Stop()

File: web/web.go [hunk 5]
```diff
@@ -201,7 +202,7 @@ func New(o *Options) *Handler {
 
 	router.Get("/heap", readyf(instrf("heap", dumpHeap)))
 
-	router.Get(o.MetricsPath, readyf(prometheus.Handler().ServeHTTP))
+	router.Get(o.MetricsPath, readyf(promhttp.Handler().ServeHTTP))
```
Comment (grobie) [inline web/web.go]: No, we can't do this here, the new HTTP handler has new semantics and is not a 1:1 replacement. This should be a 2.0 change.

File: documentation/examples/remote_storage/remote_storage_adapter/main.go [hunk 6]
```diff
@@ -94,7 +95,7 @@ func init() {
 
 func main() {
 	cfg := parseFlags()
-	http.Handle(cfg.telemetryPath, prometheus.Handler())
+	http.Handle(cfg.telemetryPath, promhttp.Handler())
```
Comment (grobie) [inline documentation/examples/remote_storage/remote_storage_adapter/main.go]: Well, it's an example, but this is a breaking change removing all the default instrumented metrics.

File: storage/local/mapper_test.go (hunk 1 above)
Comment (SuperQ) [reply storage/local/mapper_test.go]: Sure, I can fix those.

File: web/web.go (hunk 5 above)
Comment (SuperQ) [reply web/web.go]: Sounds good.

**Comment by mdlayher** [comment]
If you're concerned about the deprecation warnings, but don't want to use promhttp until 2.0, you can choose to specifically ignore those with a flag to staticcheck. It's not ideal but works.

File: storage/remote/queue_manager.go (hunk 4 above)
Comment (SuperQ) [reply storage/remote/queue_manager.go]: Going to revert this one for now, as it breaks the code:

```
queue_manager.go:256:10: invalid operation: cannot receive from non-channel ticker (variable of type *time.Ticker)
```

**Comment by SuperQ** [comment]
@mdlayher Yes, I fixed up the `-ignore` option.

**Comment by mdlayher** [comment]
@SuperQ planning to merge?

**Comment by SuperQ** [comment]
Ahh, sure, just forgot to click the button.

**Pull request merged**